lighthouse-core/lib/third-party-web.js

```diff
@@ -23,6 +24,20 @@ function getEntity(url) {
   }
 }
 
+/**
+ * `third-party-web` throws when the passed in string doesn't appear to have any domain whatsoever.
+ * We pass in some not-so-url-like things, so make the dependent-code simpler by making this call safe.
+ * @param {string} url
+ * @return {ThirdPartyProduct|undefined}
+ */
+function getProduct(url) {
+  try {
+    return thirdPartyWeb.getProduct(url);
+  } catch (_) {
+    return undefined;
```

Comment:
Were we talking about getting rid of the try block entirely? I can't find a situation where `getProduct` or `getEntity` will throw.

Reply:
> Were we talking about getting rid of the try block entirely? I can't find a situation where `getProduct` or `getEntity` will throw.

Based on the comment it sounds like they threw on invalid URLs, but maybe that changed in more recent versions of `third-party-web`?

Reply:
@patrickhulce do you know what would cause `getProduct` or `getEntity` to throw?

Reply:
originally it would throw if the provided URL was completely invalid, I [since updated it to return `null` in that case](https://github.com/patrickhulce/third-party-web/commit/fa6e9888012a29dab76c4c3f5b79619643675039#diff-cd3025771cb9802302b141b004bef6c1). conceivably there could be another unexpected case in third-party-web that throws we're not aware of, but that would be considered a bug and not really worth of a try/catch :)

Reply:
I removed both try blocks.